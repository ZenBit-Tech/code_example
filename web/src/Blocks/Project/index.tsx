import React from 'react'; // we need this to make JSX compile
import {
  ProjectWrapper,
  ProjectWrapperTop,
  ProjectWrapperBottom,
  ProjectWrapperBottomLeft,
  ProjectWrapperBottomRight,
  Title,
  Time,
  Subscriber,
} from './styles.js';

export type ProjectProps = {
  image?: string;
  title: string;
  time: string;
};

export const Project = ({ title, image, time }: ProjectProps) => (
  <ProjectWrapper>
    <ProjectWrapperTop image={image} />
    <ProjectWrapperBottom>
      <ProjectWrapperBottomLeft>
        <Title>{title}</Title>
        <Time>{time}</Time>
      </ProjectWrapperBottomLeft>
      <ProjectWrapperBottomRight>
        <Subscriber image={image} />
        <Subscriber image={image} />
        <Subscriber image={image} />
      </ProjectWrapperBottomRight>
    </ProjectWrapperBottom>
  </ProjectWrapper>
);
