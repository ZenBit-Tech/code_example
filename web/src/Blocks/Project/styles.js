import styled, { css } from 'styled-components';

export const ProjectWrapper = styled.div`
  width: 216px;
  height: 156px;
  border-radius: 6px 6px 6px 6px;
  background: #ffffff;
  &:hover {
    box-shadow: 0 0 10px rgb(0 0 0 / 50%);
  }
`;
export const ProjectWrapperTop = styled.div`
  height: 117px;
  background-size: cover;
  background-position: center;
  border-radius: 6px 6px 0px 0px;

  ${(props) =>
    css`
      background-image: url(${props.image});
    `};
`;
export const ProjectWrapperBottom = styled.div`
  border-radius: 0px 0px 6px 6px;
  display: flex;
  justify-content: space-between;
  height: calc(100% - 117px);
  align-items: center;
  padding: 0px 10px;
  filter: drop-shadow(0px 2.4px 2.4px rgba(0, 0, 0, 0.25));
`;
export const ProjectWrapperBottomLeft = styled.div`
  display: flex;
  flex-direction: column;
`;
export const ProjectWrapperBottomRight = styled.div`
  //background: green;
  height: 100%;
  width: 40%;
  display: flex;
  align-items: center;
`;
export const Title = styled.h3`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 8.4px;
  line-height: 10px;
  display: flex;
  align-items: flex-end;
  color: #000000;
  margin-bottom: 5px;
`;
export const Time = styled.p`
  font-family: Roboto;
  font-style: normal;
  font-weight: normal;
  font-size: 7.2px;
  line-height: 8px;
  display: flex;
  align-items: flex-end;
  letter-spacing: 0.078px;
  color: #666666;
`;
export const Subscriber = styled.div`
  width: 24px;
  height: 24px;
  border-radius: 100%;
  background-size: cover;
  background-position: center;
  ${(props) =>
    css`
      background-image: url(${props.image});
    `};
`;
