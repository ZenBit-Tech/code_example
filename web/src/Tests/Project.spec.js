import React from 'react';
import { create } from 'react-test-renderer';
import { Project } from 'Blocks/Project/index';

describe('Project block', () => {
  test('Project', () => {
    const projectBlock = create(
      <Project
        image="https://icatcare.org/app/uploads/2018/07/Thinking-of-getting-a-cat.png"
        title="ToolKit"
        time="2 hours ago"
      />,
    );

    const testInstance = projectBlock.root;
    expect(projectBlock.toJSON()).toMatchSnapshot();
    expect(testInstance.findByType(Project).props.title).toBe('ToolKit');
  });
});
