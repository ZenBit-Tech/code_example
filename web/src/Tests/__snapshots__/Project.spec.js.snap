// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project block Project 1`] = `
<div
  className="sc-bdfBwQ lldqtx"
>
  <div
    className="sc-gsTCUz dgdUNE"
  />
  <div
    className="sc-dlfnbm FPsoe"
  >
    <div
      className="sc-hKgILt jsOJFj"
    >
      <h3
        className="sc-jSgupP capTFj"
      >
        ToolKit
      </h3>
      <p
        className="sc-gKsewC frvdQs"
      >
        2 hours ago
      </p>
    </div>
    <div
      className="sc-eCssSg dmeamK"
    >
      <div
        className="sc-iBPRYJ lijGoy"
      />
      <div
        className="sc-iBPRYJ lijGoy"
      />
      <div
        className="sc-iBPRYJ lijGoy"
      />
    </div>
  </div>
</div>
`;
